client/rpc-api/src/state/mod.rs

```diff
@@ -136,4 +136,103 @@ pub trait StateApi<Hash> {
+	/// do not reduce time for actual block re-execution, they reduce the response payload size.
+	///
+	/// Note: storage events primarily come from _primitives/state-machine/src/ext.rs_).
+	/// The default filters can be overridden, see details in the [params section](#params).
+	///
+	/// ### Params
+	///
+	/// - `block_hash` (param index 0): Hash of the block to trace.
+	/// - `targets` (param index 1): String of comma separated (no spaces) targets. Specified
+	/// 	targets matched with trace targets by prefix (i.e if a target is in the beginning
+	/// 	of a trace target it is considered a match). If an empty string is specified no
+	/// 	targets will be filtered out. The majority of targets correspond to module names,
+	/// 	and the ones that do not are typically "hardcoded" into span or event location
+	/// 	somewhere in Substrate source. ("Non-hardcoded" targets typically come from frame
+	/// 	support macros.)
+	/// - `storage_keys` (param index 2): String of comma separated (no spaces) hex encoded
+	/// 	(no `0x` prefix) storage keys. If an empty string is specified no events will
+	/// 	be filtered out. If anything other than an empty string is specified, events
+	/// 	will be filtered by storage key (so non-storage events will **not** show up).
+	/// 	You can specify any degree of a storage key prefix (i.e. if a specified storage
+	/// 	key is in the beginning of an events storage key it is considered a match).
+	/// 	Example: for balance tracking on Polkadot & Kusama you would likely want
+	///		to track changes to account balances with the frame_system::Account storage item,
+	///		which is a map from `AccountId` to `AccountInfo`. The key filter for this would be
+	///		the storage prefix for the map:
+	///		`26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9`
+	/// 	Additionally you would want to track the extrinsic index, which is under the
+	///		`:extrinsic_index` key. The key for this would be the aforementioned string as bytes
+	///		in hex: `3a65787472696e7369635f696e646578`.
+	///		The following are some resources to learn more about storage keys in substrate:
+	///		[substrate storage][1], [transparent keys in substrate][2],
+	///		[querying substrate storage via rpc][3].
+	///
+	///		[1]: https://substrate.dev/docs/en/knowledgebase/advanced/storage#storage-map-key
+	///		[2]: https://www.shawntabrizi.com/substrate/transparent-keys-in-substrate/
+	///		[3]: https://www.shawntabrizi.com/substrate/querying-substrate-storage-via-rpc/
+	///
+	/// ### Maximum payload size
+	///
+	/// The maximum payload size allowed is 15mb. Payloads over this size will return a
+	/// object with a simple error message. If you run into issues with payload size you can
+	/// narrow down the traces using a smaller set of targets and/or storage keys.
+	///
+	/// If you are having issues with maximum payload size you can use the flag
+	/// `-lstate_tracing=trace` to get some logging during tracing.
```

Comment:
This is probably the most thoroughly documented RPC in all of substrate, great job.

Maybe add an example `curl` command?
Maybe add a link to the `substrate-archive` wasm blobs collection (as I suspect that is the trickiest step when setting this up)